package/libnetwork/src/libnetwork.h

```diff
@@ -30,9 +30,10 @@
  * @brief   Config for network.
  */
 typedef struct {
-  const char *url;  /**< Network url to connect to. */
-  int fd;           /**< File descriptor to read to or write from. */
-  bool debug;       /**< Enable debugging output */
+  const char *url;        /**< Network url to connect to. */
+  int fd;                 /**< File descriptor to read to or write from. */
+  bool debug;             /**< Enable debugging output */
+  double gga_xfer_secs;   /**< Time between NMEA GGA uploads */
```

Comment:
Ugh, kind of don't like putting this in here - will be unitialized data from the other calls now. Maybe as another param?

Reply:
Not really sure what the best path forward here is... we already have values with things that are not explicitly initialized for network_progress_t... and as the complexity of this grows (and the need to share data between various callbacks becomes present) it's less obvious what best path is here.  I'm starting to like the idea of a big "libnetwork context" structure.

To make things explicit, I tried moving things into specific structs for each callback, but this requires a pointer for the network_progress callback to read the last_xfer_time field... so, not super keen on this.

Reply:
Yeah, I guess things are getting a little hard to factor out cleanly as there's now mutable state across multiple callbacks. The uninitialized structs in network_progress_t still seems a little rough as well as in the config:

https://github.com/swift-nav/piksi_buildroot/blob/v1.3.0-release/package/skylark_download_daemon/src/skylark_download_daemon.c#L124
https://github.com/swift-nav/piksi_buildroot/blob/v1.3.0-release/package/skylark_upload_daemon/src/skylark_upload_daemon.c#L103

Not a big deal and fine for this change I guess, but might be worth refactoring things and loosing some of these structs.